### PR TITLE
Improve AOT snapshot input verification + cleanup

### PR DESCRIPTION
**Pull request by cbracken**
Bugfix: Moves AOT snapshot input verification past where the last input
is added to the inputs list.

Cleanup:
* Extracts _isValidAotPlatform method.
* Moves non-platform-specific logic to the top.
* Moves variable declaration closer to first use, and inlines to a
  narrower scope where possible.

### CODE REVIEW

**Review by tvolkert — APPROVED** [review]

File: packages/flutter_tools/lib/src/base/build.dart [hunk 1]
```diff
@@ -264,99 +264,93 @@ class Snapshotter {
...
-    final String assemblyO = fs.path.join(outputDir.path, 'snapshot_assembly.o');
-    final String assemblySo = fs.path.join(outputDir.path, 'app.so');
-    final bool compileToSharedLibrary =
-        preferSharedLibrary && androidSdk.ndkCompiler != null;
-
+    final bool compileToSharedLibrary = preferSharedLibrary && androidSdk.ndkCompiler != null;
     if (preferSharedLibrary && !compileToSharedLibrary) {
-      printStatus(
-          'Could not find NDK compiler. Not building in shared library mode');
+      printStatus('Could not find NDK compiler. Not building in shared library mode');
```
Comment (goderbauer) [inline packages/flutter_tools/lib/src/base/build.dart]: nit: finish with a full stop `.`?

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter_tools/lib/src/base/build.dart (hunk 1 above)
Comment (cbracken) [reply packages/flutter_tools/lib/src/base/build.dart]: Oh nice! Done.

**Comment by cbracken** [comment]
Related #2089

**Pull request merged**